docs/NewTargetSpecificNode.md

```diff
@@ -0,0 +1,93 @@
+
+We should take both fast operator kernel implementation and extra potential layout transformation into consideration to get better performance.
+
+### Steps
+
+There are mainly three steps to add new target specific node in Glow:
+
+1. Add a backend-specific Node `FusedAB` to `XSpecificNodes.h`, and a corresponding backend-specific Instruction `FusedAB` to `XSpecificInstrs.h`. Note that the `FusedABInst` needs to be marked with `autoIRGen()` so that the node is automatically IRGen'd to the instruction, as we currently do not support backend-specific IRGen.
+
+2. Add logic to `XBackend::transformPreLowering()` or `XBackend::transformPostLowering()` (or both) depending on when you want to do the transformation. This logic would look for the pattern of nodes you want to fuse (`A` with a single use by `B`), and replaces all uses of the result of B with the new backend-specific `FusedABNode`.
+
+3. Have your backend `X` implement `FusedABInst`. For example, for the OpenCL backend, this would mean adding a case to enqueue a kernel for the `FusedABInst` to `OpenCLFunction::execute()`, and then adding the corresponding kernel in `kernels.cl`.
+
+
+### Examples
+
+#### Operator Fusion for ReLU in CPU
+
+ReLU is max between zero and the input value. Glow lowers ReLU Node to two basic low-level linear algebra operator nodes, Splat Node and Max Node.
+
+Please refer to the document in [Backend](https://github.com/pytorch/glow/blob/master/docs/Backends.md#backend-specific-nodes-and-instructions) part for source code details on adding a new target specific CPUMaxSplatNode on CPU. 
+
+#### Data Layout Transformation for Conv Operator in OpenCL
+
+The default layout of convolution operator in Glow is `NHWC`. Since OpenCL fast convolution kernel itself has some issue with group > 1, we should convert layout `NHWC` to `NCHW` if the group > 1. Please refer to the [TODO](https://github.com/pytorch/glow/blob/master/lib/Backends/OpenCL/Transforms.cpp#L39) note. 
+
+The OpenCL backend defines `OCLConvolution` in `tools/ClassGen/OpenCL/OpenCLSpecificNodes.h` to support layout `NCHW` input.
+
+```cpp
+BB.newNode("OCLConvolution")
+    .addInput("Input")
+    .addInput("Filter")
+    .addInput("Bias")
+    .addMember(MemberType::VectorUnsigned, "Kernels")
+    .addMember(MemberType::VectorUnsigned, "Strides")
+    .addMember(MemberType::VectorUnsigned, "Pads")
+    .addMember(MemberType::Unsigned, "Group")
+    .addResultFromCtorArg()
+    .setDocstring(
+        "This is an OpenCL-specific convolution implementation where the "
+        "filter, the bias and the input are in the NCHW format");
+```
+
+During `transformPostLowering()`, this `convertConvToNCHWConv` node which contains a `NCHWConvNode` node and `Transpose` node  replaces the aforementioned pattern. 
```

Comment:
nit: Please make clear here that this optimization adds multiple transposes, on the `Input`, `Filter`, and `Result`.

Reply:
Done